### PR TITLE
Support listening on unix domain socket

### PR DESCRIPTION
**Pull request by PhracturedBlue**
This patch allows listening on  a unix-domain-socket instead of via TCP.  This is useful when using rootless containers without slirp4netns by exposing the unix socket through the host rather than mapping unix ports.  In my case, it allows me to run a reverse-proxy with --network=none as it only speaks to unix sockets, and listens on a systemd socket-activated socket.

To use, specify:
 * HTTP_BINDING_ADDRESS=<absolute path to binding socket>
 * ~PORT=0~

### CODE REVIEW

**Review by JakobLichterfeld — APPROVED** [review]
lgtm

**Comment by JakobLichterfeld** [comment]
Please add documentation as well

**Comment by PhracturedBlue** [comment]
I have pushed an update with documentation along with some improvements.  ranch >= 2.0 is needed to properly support auto-removal and permission-setting of the unix-domain-socket.

File: website/docs/guides/unix_domain_sockets.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+## Requirements
+
+- Linux system configured with Teslamate installed and working
+  - These instructions will document the procedure for using a UDS with docker-compose, but it is not difficult to adapt them to a system running Teslamate natively via systemd.
+- Nginx configured as a reverse proxy
+
+
+## Instructions
+
+Nginx requires that the UDS exist when it is started, but Teslamate will (re)create the UDS on startup.  This means that Teslamate must be configured to start before Nginx, or Nginx must be configured to detect a socket change and reload (for example the [socket-gen](https://github.com/PhracturedBlue/socket-gen) utility designed for this purpose). Additionally, because docker-compose does not provide a method to run host-commands prior to starting a container, the directory containing the UDS must be manually created before Teslamate starts.  It is easiest to manually create this directory e unix-somain-socket on a persistent volume.
```
Comment (JakobLichterfeld) [inline website/docs/guides/unix_domain_sockets.md:17]: "This directory e unix-somain-socket" ?

File: website/docs/guides/unix_domain_sockets.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+---
+title: Using Unix Domain Sockets with a reverse-proxy
+---
+
+It is possible to configure Teslamate to communicate over unix-domain sockets (UDS) instea of a typical network socket.  This can be useful to improve security by restricting which applications can communicate to the application.  A typical configuration would be to use a UDS between a reverse-proxy (like Nginx) and Teslamate.  When paired with something like rootless-podman and [socket-activation](https://github.com/containers/podman/blob/main/docs/tutorials/socket_activation.md), Nginx can be configured with `--network=none` providing external access to Teslmate without the Nginx container having any networking at all. While setting up socket-activation and Podman is eyond the scope of this document, it will explain how to configure UDS between Teslamate and an Nginx reverse-proxy.
```
Comment (JakobLichterfeld) [inline website/docs/guides/unix_domain_sockets.md:5]: Should be "**b**eyond the scope"

**Review by JakobLichterfeld — CHANGES_REQUESTED** [review]
Thanks for the change and the documentation!

File: website/docs/guides/unix_domain_sockets.md (hunk 2 above)
Comment (PhracturedBlue) [reply website/docs/guides/unix_domain_sockets.md]: Thanks for looking it over.  Fixed

**Review by JakobLichterfeld — APPROVED** [review]

**Pull request merged**